src/sourceItemWrapper.js

```diff
@@ -663,32 +663,33 @@ class SourceItemWrapper extends ItemWrapper {
                 'loading',
                 Wikicite.getString('wikicite.source-item.add-identifier.progress.loading')
             );
-
             try {
                 if (identifiers.length > 0) {
-                    let citations = [];
                     await Zotero.Schema.schemaUpdatePromise;
-                    for (const identifier of identifiers) {
-                        let translation = new Zotero.Translate.Search();
+                    let citations = await Promise.all(identifiers.map(async (identifier) => {
+                        var translation = new Zotero.Translate.Search();
                         translation.setIdentifier(identifier);
+                        let translators = await translation.getTranslators();
+                        translation.setTranslator(translators);
 
                         let jsonItems;
                         try {
                             // set libraryID to false so we don't save this item in the Zotero library
                             jsonItems = await translation.translate({libraryID: false});
-                        } catch {
```

Comment:
`translation.translate()` will throw a "No items returned from any translator" if no items were found for the identifier provided. This behavior is expected and we don't want to bother the user with error messages in these cases, just skip the identifier. That's why we were catching errors on `translation.translate()` alone.
The rest of the function below is not expected to fail, though. If it does, we do want to throw an error so the whole operation fails.
If my reasoning is correct, I'd rather continue catching errors only on `translation.translate()`